cmd/juju/application/updateseries.go

```diff
@@ -109,32 +117,47 @@ func (c *updateSeriesCommand) Init(args []string) error {
 // Run implements cmd.Run.
 func (c *updateSeriesCommand) Run(ctx *cmd.Context) error {
 
-	if c.updateSeriesClient == nil {
+	if c.updateApplicationSeriesClient == nil && c.updateMachineSeriesClient == nil {
 		apiRoot, err := c.NewAPIRoot()
 		if err != nil {
 			return errors.Trace(err)
 		}
 		defer apiRoot.Close()
-		c.updateSeriesClient = application.NewClient(apiRoot)
-		defer c.updateSeriesClient.Close()
+		c.updateApplicationSeriesClient = application.NewClient(apiRoot)
```

Comment:
Don't we only want to make the client depending on what type of arg is passed? ie if they specify a machine, we don't need to create an application facade client and visa versa

Reply:
Updated.